### PR TITLE
Error message if TypeScript version not ok - fix #88

### LINKED ISSUE #88: TypeError: tsModule.formatDiagnosticsWithColorAndContext is not a function
#### What happens and why it is wrong

complie error.
TypeError: tsModule.formatDiagnosticsWithColorAndContext is not a function

#### Versions
- typescript: typescript@2.8.4
- rollup: rollup@0.59.2 
- rollup-plugin-typescript2: rollup-plugin-typescript2@0.15.0

#### rollup.config.js

```js
import resolve from 'rollup-plugin-node-resolve';
import commonjs from 'rollup-plugin-commonjs';
import typescript from 'rollup-plugin-typescript2';
import serve from 'rollup-plugin-serve';
import livereload from 'rollup-plugin-livereload';
import { uglify } from 'rollup-plugin-uglify';

const config = require('../config/index');
const env = require('../util/env');

let fileType = '';
if (env.isProd) {
    if (env.isEs) {
        fileType = '.esm';
    } else {
        fileType = '.min';
    }
}

let format = 'umd';
if (env.isEs) {
    format = 'es';
}

const rollupConfig = {
    input: './src/index.ts',
    output: {
        file: `dist/${config.name}${fileType}.js`,
        name: config.name,
        format,
        globals: {

        },
        sourcemap: true,
    },
    watch: {
        exclude: 'node_modules/**',
    },
    plugins: [
        typescript(),
        resolve(),
        commonjs(),
    ],
};

if (env.isDev) {
    rollupConfig.plugins.push(serve({
        open: config.dev.autoOpenBrowser,
        contentBase: ['dist', 'demo'],
        historyApiFallback: false,
        host: config.dev.host,
        port: config.dev.port,
    }));
    rollupConfig.plugins.push(livereload());
}
if (env.isProd) {
    rollupConfig.plugins.push(uglify());
}

export default rollupConfig;
```

#### tsconfig.json

```json
{
    "compilerOptions": {
        "module": "es2015",
        "lib": [
            "dom",
            "es2015"
        ]
    }
}
```

#### plugin output with verbosity 3

```
[!] TypeError: tsModule.formatDiagnosticsWithColorAndContext is not a function
TypeError: tsModule.formatDiagnosticsWithColorAndContext is not a function
    at /Users/yinxin/github/damai-cli/node_modules/_rollup-plugin-typescript2@0.15.0@rollup-plugin-typescript2/src/tscache.ts:56:25
    at arrayMap (/Users/yinxin/github/damai-cli/node_modules/_rollup-plugin-typescript2@0.15.0@rollup-plugin-typescript2/node_modules/lodash/lodash.js:639:23)
    at map (/Users/yinxin/github/damai-cli/node_modules/_rollup-plugin-typescript2@0.15.0@rollup-plugin-typescript2/node_modules/lodash/lodash.js:9554:14)
    at convertDiagnostic (/Users/yinxin/github/damai-cli/node_modules/_rollup-plugin-typescript2@0.15.0@rollup-plugin-typescript2/src/tscache.ts:52:3)
    at Object.options (/Users/yinxin/github/damai-cli/node_modules/_rollup-plugin-typescript2@0.15.0@rollup-plugin-typescript2/src/index.ts:134:27)
    at applyOptionHook (/Users/yinxin/github/damai-cli/node_modules/_rollup@0.59.2@rollup/dist/rollup.js:21107:23)
    at Array.reduce (<anonymous>)
    at getInputOptions$1 (/Users/yinxin/github/damai-cli/node_modules/_rollup@0.59.2@rollup/dist/rollup.js:21124:33)
    at Object.rollup (/Users/yinxin/github/damai-cli/node_modules/_rollup@0.59.2@rollup/dist/rollup.js:21128:30)
    at build (/Users/yinxin/github/damai-cli/node_modules/_rollup@0.59.2@rollup/bin/rollup:3444:19)
```

### PR DESCRIPTION
**Pull request by mathiasrw**
Before building the string indicating valid range for a TypeScript version will be placed into `src/` files. 

If the installed typescript version is not OK the user will get a useful message:

skaermbillede 2018-06-20 kl 01 41 34

---
This change is [Reviewable](https://reviewable.io/reviews/ezolenko/rollup-plugin-typescript2/92)

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -23,7 +23,7 @@
   "homepage": "https://github.com/ezolenko/rollup-plugin-typescript2",
   "author": "@ezolenko",
   "scripts": {
-    "prebuild": "rimraf dist/*",
+    "prebuild": "rimraf dist/* && rexreplace '(requredTypescriptRange = .)[^\"]+' '€1+require(process.cwd()+\"/package.json\").peerDependencies.typescript' -j src/*.ts",
```
Comment (ezolenko) [inline package.json]: This is better done in `rollup.config.base.js`, there is already `rollup-plugin-re` invocation in there. See how `$RPT2_VERSION` is treated.

File: package.json [hunk 2]
```diff
@@ -40,26 +40,29 @@
     "typescript": ">=2.4.0"
```
Comment (ezolenko) [inline package.json]: Don't need that anymore. Might want to rebase the PR on master to avoid extra changes.

**Comment by mathiasrw** [comment]
Thanks for the feedback. I will look into this during the weekend.

**Comment by ezolenko** [comment]
Changes are adapted and committed in master

**Pull request closed without merging**